File: leetcode/archive/contains-duplicate/contains_duplicate.cpp
#include <iostream>
#include <vector>
#include <unordered_set>

class Solution {
    public:
        bool containsDuplicate(std::vector<int> &nums) {
            std::unordered_set<int> set;
            for (int n : nums) {
                if (!set.insert(n).second) {
                    return true;
                }
            }
            return false;
    }
};

int main() {
    return 0;
}

File: leetcode/archive/contains-duplicate/ContainsDuplicate.java
import java.util.HashSet;
import java.util.Set;

public class ContainsDuplicate {
    public static void main(String[] args) {
        
    }
}

class Solution {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for (int n : nums) {
            if (!set.add(n)) {
                return true;
            }
        }
        return false;
    }
}